SDWebImageFLPlugin/Classes/FLAnimatedImageBridge/SDWebImageFLCoder.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * This file is part of the SDWebImage package.
+ * (c) Olivier Poitrey <rs@dailymotion.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+#import <SDWebImage/SDWebImage.h>
+
+// A coder which decode the GIF image, into `FLAnimatedImage` representation and bind the associated object. See `UIImage+SDWebImageFLPlugin` for more detailed information.
+// When you want to use `FLAnimatedImageView` to load image, be sure to add this coder before `SDImageGIFCoder`, to ensure this coder been processed before `SDImageGIFCoder`
+
+@interface SDWebImageFLCoder : NSObject <SDImageCoder>
```

Comment:
Just some thoughts after I looked coder mechanism.
The coder mechanism is sufficient currently? Maybe we can add the url parameter and a context option which can specify the coder class per image.

Reply:
@zhongwuzw You can pass anything into context, and get the context back. This is a football design (The context arg pass from really top-level API to the bottom).

Actually it's a bad design. A coder, it's designed to solve the image decoding task, but not any business logic related to things other than decoding. However, we leave it as a hackable space, to enable some advanced user use this to do extra logic.

All built-in coders, should not touch `SDWebImageContext` during image decoding, if new option is requsted, update `SDImageOptions`

Reply:
@dreampiggy Emm, I mean bring the context ahead to `- (BOOL)canDecodeFromData:(NSData *)data`, which can determine wether or not can handle image in advance.

Reply:
@zhongwuzw That too much. Your image coder should only judge the image format and choose  to response it or not. Like Image/IO using UTType to determine which underneath plugin (AppleJPEGPlugin ? PNGPlugin ?) should use. So it should be been passed with the `SDImageCoderOptions` arg.

Reply:
@dreampiggy fine, no big difference.